tests/integration/test_exodus.py

```diff
@@ -0,0 +1,98 @@
+import requests
+import pytest
+
+
+def test_exodus_basic(cdn_test_url):
+    url = (
+        cdn_test_url
+        + "/content/beta/rhel8/8/x86_64/exodus_test/os/bash-4.4.19-7.el8.x86_64.rpm"
+    )
+
+    r = requests.get(url)
+    assert r.status_code == 200
+    assert "cache-control" not in r.headers
+
+
+def test_header_not_exist_file(cdn_test_url):
+    url = (
+        cdn_test_url
+        + "/content/beta/rhel8/8/x86_64/exodus_test/os/bash-4.4.19-7.el8.x86_64.rpm_no"
+    )
+    r = requests.get(url)
+    assert r.status_code == 404
+    assert "cache-control" not in r.headers
+
+
+# Test data for exdous-lambda
+# serving yum repodata with correct Content-Type
+testdata_cache_control_path = [
+    "/content/beta/rhel8/8/x86_64/exodus_test/os/listing",
+    "/content/beta/rhel8/8/x86_64/exodus_test_ctype/os/repodata/repomd.xml",
+    "/content/beta/rhel8/8/x86_64/exodus_test/os/ostree/repo/refs/heads/sub_dir/testing",
+]
+
+
+@pytest.mark.parametrize("testdata_path", testdata_cache_control_path)
+def test_header_cache_control(cdn_test_url, testdata_path):
+    url = cdn_test_url + testdata_path
+    r = requests.get(url)
+    assert r.status_code == 200
+    assert r.headers["cache-control"] == "max-age=600"
+
+
+def test_header_want_digest_GET(cdn_test_url):
+    headers = {"want-digest": "id-sha-256"}
+    url = cdn_test_url + "/content/beta/rhel8/8/x86_64/exodus_test/os/listing"
+    r = requests.get(url, headers=headers)
+    assert r.status_code == 200
+    assert (
+        r.headers["digest"]
+        == "id-sha-256=jbuWD7I1f2C3IcJT6acQqIz1kI/pffRTD3krSieXQQc="
+    )
+
+
+def test_header_want_digest_HEAD(cdn_test_url):
+    headers = {"want-digest": "id-sha-256"}
+    url = cdn_test_url + "/content/beta/rhel8/8/x86_64/exodus_test/os/listing"
+    r = requests.head(url, headers=headers)
+    assert r.status_code == 200
+    assert (
+        r.headers["digest"]
+        == "id-sha-256=jbuWD7I1f2C3IcJT6acQqIz1kI/pffRTD3krSieXQQc="
+    )
+
+
+# Test data for exodus-lambda
+# serving repo entry points with appropriate cache headers
```

Comment:
Similarly this comment refers to something different from what's tested...

Did two comments get swapped around?